lib/katello/tasks/repository.rake

```diff
@@ -0,0 +1,101 @@
+    repos
+  end
+
+  def lookup_puppet_environments
+    lifecycle_envs = Katello::KTEnvironment.where(:name => ENV['LIFECYCLE_ENVIRONMENT']) if ENV['LIFECYCLE_ENVIRONMENT']
+    content_views = Katello::ContentView.where(:name => ENV['CONTENT_VIEW']) if ENV['CONTENT_VIEW']
+
+    repos = ::Katello::ContentViewPuppetEnvironment
+    repos = repos.in_environment(lifecycle_envs) if lifecycle_envs
+    repos = repos.in_content_view(content_views) if content_views
+    repos
+  end
+
+  desc "Regnerate metadata for all repositories. Specify CONTENT_VIEW=name and LIFECYCLE_ENVIRONMENT=name to narrow repositories."
+  task :regenerate_repo_metadata => ["environment", "check_ping"] do
+    User.current = User.anonymous_api_admin
+    repos = lookup_repositories
+
+    if repos.any?
+      task = ForemanTasks.async_task(::Actions::BulkAction, Actions::Katello::Repository::MetadataGenerate, repos.all.sort)
+      puts "Regenerating #{repos.count} repositories.  You can monitor these on task id #{task.id}\n"
+    else
+      puts "No repositories found for regeneration."
+    end
+  end
+
+  desc "Refresh repository metadata for all repositories. Specify CONTENT_VIEW=name and LIFECYCLE_ENVIRONMENT=name to narrow repositories."
+  task :refresh_pulp_repo_details => ["environment", "check_ping"] do
+    User.current = User.anonymous_api_admin
+    repos = lookup_repositories
+
+    if repos.any?
+      task = ForemanTasks.async_task(::Actions::BulkAction, Actions::Katello::Repository::RefreshRepository, repos.all.sort)
+      puts "Refreshing #{repos.count} repositories.  You can monitor these on task id #{task.id}\n"
+    else
+      puts "No repositories found for regeneration."
+    end
+  end
+
+  def repo_exists?(repo)
+    Katello.pulp_server.extensions.repository.retrieve(repo.pulp_id)
+    true
+  rescue RestClient::ResourceNotFound
+    false
+  end
+
+  def handle_missing_repo(repo)
+    commit = ENV['COMMIT'] == 'true'
+    puts "Repository #{repo.id} Missing"
+    if repo.content_view.default?
+      puts "Recreating #{repo.id}"
+      ForemanTasks.sync_task(::Actions::Katello::Repository::Create, repo) if commit
+    else
+      puts "Deleting #{repo.id}"
+      ForemanTasks.sync_task(::Actions::Katello::Repository::Destroy, repo, :planned_destroy => true) if commit
+    end
+  end
+
+  def handle_missing_puppet_env(puppet_env)
+    puts "Content View Puppet Environment #{puppet_env.id} Missing, Creating."
+    ForemanTasks.sync_task(::Actions::Katello::ContentViewPuppetEnvironment::Create, puppet_env) if ENV['COMMIT'] == 'true'
+  end
+
+  desc "Correct missing pulp repositories. Specify CONTENT_VIEW=name and LIFECYCLE_ENVIRONMENT=name to narrow repositories.  COMMIT=true to perform operation."
+  task :correct_repositories => ["environment", "check_ping"] do
+    puts "All operations will be skipped.  Re-run with COMMIT=true to perform corrections." if ENV['COMMIT'] != 'true'
```

Comment:
which other tasks use NOOP?  looking at katello i see 1 other task use COMMIT, but none use NOOP. 

 I kinda prefer COMMIT, as to me it makes more sense to say 'YES commit these changes' rather than 'Yes now run in NOOP false mode'.  I think part of my hangup with NOOP, is you're saying 'disable' this thing, rather than 'enable this thing' to actually make a change, but that might be personal preference.